content/blog/new-per-subject-discard-policy.md

```diff
@@ -0,0 +1,223 @@
+
+You can select what you want for the other parameters, or just hit return to use the defaults, as long as you set the "Per subject Messages Limit" to `1`, and the stream's discard policy to `new` (as this new discard per subject is not relevant to the `old` stream discard policy).
+```
+? Subjects foo.*
+? Storage file
+? Replication 1
+? Retention Policy Limits
+? Discard Policy New
+? Stream Messages Limit -1
+? Per Subject Messages Limit 1
+? Total Stream Size -1
+? Message TTL -1
+? Max Message Size -1
+? Duplicate tracking time window 0
+? Allow message Roll-ups No
+? Allow message deletion Yes
+? Allow purging subjects or the entire stream Yes
+Stream dedup was created
+
+Information for Stream dedup created 2022-11-16 16:57:54
+
+             Subjects: foo.*
+             Replicas: 1
+              Storage: File
+
+Options:
+
+            Retention: Limits
+     Acknowledgements: true
+       Discard Policy: New Per Subject
+     Duplicate Window: 2m0s
+    Allows Msg Delete: true
+         Allows Purge: true
+       Allows Rollups: false
+
+Limits:
+
+     Maximum Messages: unlimited
+  Maximum Per Subject: 1
+        Maximum Bytes: unlimited
+          Maximum Age: unlimited
+ Maximum Message Size: unlimited
+    Maximum Consumers: unlimited
+
+
+State:
+
+             Messages: 0
+                Bytes: 0 B
+             FirstSeq: 0
+              LastSeq: 0
+     Active Consumers: 0
+```
+Let's do a JetStream publish of a message for id=1
+```
+nats req foo.1 one
+``` 
+Which is successful (here the message is assigned sequence number 1):
+```
+17:08:57 Sending request on "foo.1"
+17:08:57 Received with rtt 284.958µs
+{"stream":"dedup", "seq":1}
+```
+Let's publish another message, for id=2
+```
+nats req foo.2 one
+```
+Which is also successful (and assigned sequence number 2):
+```
+17:09:03 Sending request on "foo.2"
+17:09:03 Received with rtt 267.584µs
+{"stream":"dedup", "seq":2}
+```
+Now let's try to publish another message for id=1
+```
+nats req foo.1 two
+```
+Which fails, thereby ensuring message deduplication:
+```
+17:09:11 Sending request on "foo.1"
+17:09:11 Received with rtt 192.709µs
+{"error":{"code":503,"err_code":10077,"description":"maximum messages per subject exceeded"},"stream":"dedup","seq":0}
+```
+At this point we can for example remove the existing message for id=2 using its sequence number (2 in this example)
+```
+nats stream rmm dedup 2
+```
+And attempt to publish a new message for id=2
+```
+nats req foo.2 two
+```
+Which works:
+```
+17:09:52 Sending request on "foo.2"
+17:09:52 Received with rtt 318.125µs
+{"stream":"dedup", "seq":3}
+```
+
+## Closing thoughts
+
+To resume: the newly introduced discard policy `new` for per subject limit now enables the ability to have an 'infinite deduplication window' to avoid duplicated publications that can otherwise happen to some failure scenarios.
```

Comment:
```suggestion
To review, the new discard policy *discard-new-per-subject* enables the ability to have an 'infinite deduplication window' to avoid duplicated publications that can otherwise happen to some failure scenarios.
```